generators/client/templates/angular/src/main/webapp/app/shared/fields/base-autocomplete.component.ts.ejs

```diff
@@ -0,0 +1,52 @@
+<%#
+ Copyright 2013-2018 the original author or authors from the JHipster project.
+
+ This file is part of the JHipster project, see http://www.jhipster.tech/
+ for more information.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+-%>
+
+export class BaseAutocompleteField<T> extends BaseFormField<T> {
+    searching = false;
+    searchFailed = false;
+    selected: T;
+    hideSearchingWhenUnsubscribed = new Observable(() => () => this.searching = false);
+
+    constructor(
+            config: NgbTypeaheadConfig
+        ) {
+            super();
+            config.showHint = true;
+        }
+
+    search = (text$: Observable<string>) =>
+        text$
+            .debounceTime(300)
+            .distinctUntilChanged()
+            .do(() => this.searching = true)
+            .switchMap((term) =>
+                this.<%= entityInstance %>Service.query({'<%= autocompleteField %>.contains': term})
```

Comment:
can this service be generalized and moved to the lib, the only dependency is entitySevice, cant we pass it somehow?